### PR TITLE
Reduce flakes and better test output

### PR DESCRIPTION
**Pull request by johnbelamaric**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
/kind flake

**What this PR does / why we need it**:
* Enable tests to keep running even if one test script fails
* Add a PASS and FAIL so it shows up in the Prow page
* Add a test summary for all e2e test scripts
* Fix two sources of flakes in 009.sh
* make fmt

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

/hold

### CODE REVIEW

File: e2e/lib/testing.sh [hunk 1]
```diff
@@ -23,10 +23,20 @@ function testing_run_test {
...
+    if [[ $rc != 0 ]]; then
+        result="FAIL ($rc)"
+    fi
 
-    echo "Porch Controller logs"
-    kubectl logs deployment/porch-controllers -n porch-system --since "$(($(date +%s) - int_start))s" | sed -e '/PackageVariant/!d;/resources changed/!d'
+    echo "+++++ $(date): finished $testfile $testname (result: $result)"
+    local seconds="$(($(date +%s) - int_start))"
+    printf "TIME $(basename $testfile): %s secs\n" $seconds
+    test_summary="${test_summary-}$(echo && echo $(basename $testfile): $result in $seconds seconds)"
```
Comment (electrocucaracha) [inline e2e/lib/testing.sh:36]: You can also concatenate values and print them with `echo -e $test_summary`
```suggestion
    test_summary+="$(basename $testfile): $result in $seconds seconds\n"
```

Comment (johnbelamaric) [reply e2e/lib/testing.sh:36]: "\n" doesn't work in bash, what's the better alternative? I have this:
```
[hi on] jbelamaric@jbelamaric:~/proj/gh/johnbelamaric/nephio-test-infra/e2e$ unset test
[hi on] jbelamaric@jbelamaric:~/proj/gh/johnbelamaric/nephio-test-infra/e2e$ test+="$(basename /tmp/foo): testing\n" && echo "$test"
foo: testing\n
[hi on] jbelamaric@jbelamaric:~/proj/gh/johnbelamaric/nephio-test-infra/e2e$ test+="$(basename /tmp/foo): testing\n" && echo "$test"
foo: testing\nfoo: testing\n
[hi on] jbelamaric@jbelamaric:~/proj/gh/johnbelamaric/nephio-test-infra/e2e$ test+="$(echo && echo $(basename /tmp/foo): testing)" && echo "$test"
foo: testing\nfoo: testing\n
foo: testing
[hi on] jbelamaric@jbelamaric:~/proj/gh/johnbelamaric/nephio-test-infra/e2e$ test+="$(echo && echo $(basename /tmp/foo): testing)" && echo "$test"
foo: testing\nfoo: testing\n
foo: testing
foo: testing
[hi on] jbelamaric@jbelamaric:~/proj/gh/johnbelamaric/nephio-test-infra/e2e$ test+="$(echo && echo $(basename /tmp/foo): testing)" && echo "$test"
foo: testing\nfoo: testing\n
foo: testing
foo: testing
foo: testing
[hi on] jbelamaric@jbelamaric:~/proj/gh/johnbelamaric/nephio-test-infra/e2e$ 
```
but I agree that's awkward. I swear in /sh or maybe it was csh or ksh or something, \n used to work years ago...

Comment (electrocucaracha) [reply e2e/lib/testing.sh:36]: It works if you use `-e` flag in the `echo` instruction. 

```bash
$ unset test
$ test+="$(basename /tmp/foo): testing\n" && echo -e "$test"
foo: testing

$ test+="$(basename /tmp/foo): testing\n" && echo -e "$test"
foo: testing
foo: testing

```

**Comment by johnbelamaric** [comment]
/hold cancel

I think this can be merged now.

File: e2e/tests/009.sh [hunk 2]
```diff
@@ -51,33 +51,72 @@ echo "Current Memory $current_memory"
...
+smf_pkg_rev=$smf_deployment_pkg
+
+if [[ $lifecycle == "Published" ]]; then
+    echo "Copying $smf_deployment_pkg"
+    smf_pkg_rev=$(kpt alpha rpkg copy -n default $smf_deployment_pkg --workspace $ws | cut -d ' ' -f 1)
+    echo "Copied to $smf_pkg_rev, pulling"
+fi
+
+# We need to put this entire section in a retry loop, because it is possible
+# for a controller to come in and chnage the package after we pull it. This
```
Comment (electrocucaracha) [inline e2e/tests/009.sh:68]: nit 
```suggestion
# for a controller to come in and change the package after we pull it. This
```

**Review by electrocucaracha — COMMENTED** [review]
Nice to see some execution metrics

| Task                                                | Time     |
|-----------------------------------------------------|----------|
| Deploy regional   workload cluster                  | 125 secs |
| Deploy edge   clusters                              | 550 secs |
| Deploy free5gc-cp   to regional workload cluster    | 138 secs |
| Deploy free5gc   operator to all workload clusters  | 100 secs |
| Deploy free5gc UPF   to edge clusters               | 267 secs |
| Deploy free5gc AMF   and SMF to regional clusters   | 346 secs |
| Deploy ueransim to   edge clusters                  | 318 secs |
| Vertically Scale   free5gc UPF in Edge Clusters     | 325 secs |
| Vertically Scale   free5gc SMF in Regional Clusters | 29 secs  |

So, in other words, when the modified var is set true that means that another process has changed something in the package?

**Comment by johnbelamaric** [comment]
> So, in other words, when the modified var is set true that means that another process has changed something in the package?

Yeah. One of the controllers is coming in and creating a new Draft sometimes. There are no actual deltas between it and the published version, so I don't fully understand why.
